File: itv/JCZhang-ITVConsultas-Funcional-JAVA/src/main/java/models/Coche.java
package models;

public class Coche extends Vehicle{
    int numPuertas;

    public Coche(String marca, int fabricationYear, boolean apto, double numKm, int numPuertas) {
        super(marca, fabricationYear, apto, numKm);
        this.marca = marca;
        this.fabricationYear = fabricationYear;
        this.apto = apto;
        this.numKm = numKm;
        this.numPuertas = numPuertas;
    }

    @Override
    public String toString() {
        return "Coche{" +
                "marca='" + marca + '\'' +
                ", fabricationYear=" + fabricationYear +
                ", apto=" + apto +
                ", numKm=" + numKm +
                ", numPuertas=" + numPuertas +
                '}';
    }

    public int getNumPuertas() {
        return numPuertas;
    }
}

File: itv/JCZhang-ITVConsultas-Funcional-Kotlin/src/main/kotlin/models/Coche.kt
package models

class Coche(marca: String, fabricationYear: Int, apto: Boolean, numKm: Double, val numPuertas:Int): Vehicles(marca, fabricationYear, apto,
    numKm
) {
    override fun toString(): String {
        return "Coche(marca='$marca', fabricationYear='$fabricationYear', apto=$apto, numKm=$numKm, numPuertas = $numPuertas)"
    }
}